integration_tests/tests/rpc/test_block_number.py

```diff
@@ -55,44 +74,94 @@ def test_get_block_by_block_number(client: RpcClient):
     # block number: x  | x+1 | x+2 | x+3 | x+4 |
     # epoch number: y  | y+1 | y+2 |   y + 3   |
 
-    block_0 = client.block_by_epoch("latest_mined")['hash']
-    block_a = client.generate_custom_block(parent_hash = block_0, referee = [], txs = [])
-    block_b = client.generate_custom_block(parent_hash = block_0, referee = [], txs = [])
-    block_c = client.generate_custom_block(parent_hash = block_a, referee = [], txs = [])
-    block_d = client.generate_custom_block(parent_hash = block_c, referee = [block_b], txs = [])
+    block_0 = client.block_by_epoch("latest_mined")["hash"]
+    block_a = client.generate_custom_block(parent_hash=block_0, referee=[], txs=[])
+    block_b = client.generate_custom_block(parent_hash=block_0, referee=[], txs=[])
+    block_c = client.generate_custom_block(parent_hash=block_a, referee=[], txs=[])
+    block_d = client.generate_custom_block(
+        parent_hash=block_c, referee=[block_b], txs=[]
+    )
 
-    epoch_0 = int(client.block_by_hash(block_0)['height'], 16)
-    block_number_0 = int(client.block_by_hash(block_0)['blockNumber'], 16)
+    epoch_0 = int(client.block_by_hash(block_0)["height"], 16)
+    block_number_0 = int(client.block_by_hash(block_0)["blockNumber"], 16)
 
     # make sure transactions have been executed
     parent_hash = block_d
 
     for _ in range(5):
-        block = client.generate_custom_block(parent_hash = parent_hash, referee = [], txs = [])
+        block = client.generate_custom_block(
+            parent_hash=parent_hash, referee=[], txs=[]
+        )
         parent_hash = block
 
     # check blocks by block number
     assert_equal(
-        client.block_by_block_number(hex(block_number_0)),
-        client.block_by_hash(block_0)
+        client.block_by_block_number(hex(block_number_0)), client.block_by_hash(block_0)
     )
 
     assert_equal(
         client.block_by_block_number(hex(block_number_0 + 1)),
-        client.block_by_hash(block_a)
+        client.block_by_hash(block_a),
     )
 
     assert_equal(
         client.block_by_block_number(hex(block_number_0 + 2)),
-        client.block_by_hash(block_c)
+        client.block_by_hash(block_c),
     )
 
     assert_equal(
         client.block_by_block_number(hex(block_number_0 + 3)),
-        client.block_by_hash(block_b)
+        client.block_by_hash(block_b),
     )
 
     assert_equal(
         client.block_by_block_number(hex(block_number_0 + 4)),
-        client.block_by_hash(block_d)
+        client.block_by_hash(block_d),
+    )
+
+
+# If this errors is changed, please let me know https://github.com/Conflux-Chain/rpc-errors/issues/new
+def test_get_block_by_epoch_number_errors(client: RpcClient):
+
+    try:
+        client.block_by_epoch(sys.maxsize)
```

Comment:
Code didn't raise error if no error is created. 2 alternative solutions

1. 
```py
def test_value_error():
    with pytest.raises(ValueError) as exc_info:
        raise ValueError("Invalid value")
    assert exc_info.type is ValueError
    assert str(exc_info.value) == "Invalid value"
```

2.  all use  `assert_raises_rpc_error`